logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/remote/ClientRemoteRepository.kt

```diff
@@ -1,81 +1,46 @@
 package com.wire.kalium.logic.data.client.remote
 
-import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.NetworkFailure
 import com.wire.kalium.logic.data.client.Client
 import com.wire.kalium.logic.data.client.ClientMapper
 import com.wire.kalium.logic.data.client.DeleteClientParam
 import com.wire.kalium.logic.data.client.RegisterClientParam
 import com.wire.kalium.logic.data.conversation.ClientId
-import com.wire.kalium.logic.failure.ClientFailure
 import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.wrapApiRequest
 import com.wire.kalium.network.api.user.client.ClientApi
-import com.wire.kalium.network.exceptions.KaliumException
-import com.wire.kalium.network.utils.NetworkResponse
-import com.wire.kalium.network.utils.isSuccessful
-
 
 interface ClientRemoteRepository {
-    suspend fun registerClient(param: RegisterClientParam): Either<CoreFailure, Client>
-
-    suspend fun deleteClient(param: DeleteClientParam): Either<CoreFailure, Unit>
-
-    suspend fun fetchClientInfo(clientId: ClientId): Either<CoreFailure, Client>
-
-    suspend fun fetchSelfUserClients(): Either<CoreFailure, List<Client>>
+    suspend fun registerClient(param: RegisterClientParam): Either<NetworkFailure, Client>
+    suspend fun deleteClient(param: DeleteClientParam): Either<NetworkFailure, Unit>
+    suspend fun fetchClientInfo(clientId: ClientId): Either<NetworkFailure, Client>
+    suspend fun fetchSelfUserClients(): Either<NetworkFailure, List<Client>>
 }
 
 class ClientRemoteDataSource(
     private val clientApi: ClientApi,
     private val clientMapper: ClientMapper
 ) : ClientRemoteRepository {
 
-    override suspend fun registerClient(param: RegisterClientParam): Either<CoreFailure, Client> {
-        val response = clientApi.registerClient(clientMapper.toRegisterClientRequest(param))
-        return if (response.isSuccessful())
-            Either.Right(clientMapper.fromClientResponse(response = response.value))
-        else
-            handleFailedApiResponse(response)
-    }
+    // TODO: handle too many clients in the use case
```

Comment:
This implemented and it's getting deleted by this PR? If so I think we should implement and not leave it as a TODO.

Reply:
i will implement it in this PR, i left a TODO cuz i was not sure if this should be done in `Repository` or `UseCase`

Reply:
I see